diff --git a/src/google_drive/api/files_api.cr b/src/google_drive/api/files_api.cr
index 816f0a9..e312623 100644
--- a/src/google_drive/api/files_api.cr
+++ b/src/google_drive/api/files_api.cr
@@ -2910,7 +2910,7 @@ module GoogleDrive
       # HTTP header "Accept" (if needed)
       header_params["Accept"] = @api_client.select_header_accept(["application/json"])
       # HTTP header "Content-Type"
-      header_params["Content-Type"] = @api_client.select_header_content_type(["multipart/form-data"])
+      header_params["Content-Type"] = upload_type == "media" ? MIME.from_filename(media.not_nil!.path) : @api_client.select_header_content_type(["multipart/form-data"])

       # cookie parameters
       cookie_params : Hash(String, String) = Hash(String, String).new
@@ -2935,12 +2935,12 @@ module GoogleDrive
       query_params["useContentAsIndexableText"] = use_content_as_indexable_text.to_s if !use_content_as_indexable_text.nil?

       # form parameters
-      form_params : Array(Tuple(String, Crest::ParamsValue)) | Nil = Array(Tuple(String, Crest::ParamsValue)).new
-      form_params << Tuple(String, Crest::ParamsValue).new("Media", media) if !media.nil?
-      form_params << Tuple(String, Crest::ParamsValue).new("Metadata", metadata) if !metadata.nil?
+      form_params : Array(Tuple(String, Crest::ParamsValue)) | Nil = upload_type == "multipart" ? Array(Tuple(String, Crest::ParamsValue)).new : nil
+      form_params << Tuple(String, Crest::ParamsValue).new("Metadata.json", metadata) if form_params && !metadata.nil? && upload_type == "multipart"
+      form_params << Tuple(String, Crest::ParamsValue).new("Media", media) if form_params && !media.nil? && upload_type == "multipart"

       # http body (model)
-      post_body : IO | String | Nil = nil
+      post_body : IO | String | Nil = upload_type == "multipart" ? nil : media.not_nil!

       # auth_names
       auth_names = ["Oauth2c", "Oauth2"]
